pyat/at/lattice/variable_elements.py

```diff
@@ -1,141 +1,280 @@
-              ``Amplitude(A,B)`` has to be provided
-            * For ``mode=ACMode.ARBITRARY`` the ``Func(A,B)`` corresponding to the
-              ``Amplitude(A,B)`` has to be provided
         """
-        kwargs.setdefault('PassMethod', 'VariableThinMPolePass')
-        self.MaxOrder = kwargs.pop('MaxOrder', 0)
-        self.Periodic = kwargs.pop('Periodic', True)
-        self.Mode = int(mode)
-        if AmplitudeA is None and AmplitudeB is None:
-            raise AtError('Please provide at least one amplitude for A or B')
-        AmplitudeB = self._set_params(AmplitudeB, mode, 'B', **kwargs)
-        AmplitudeA = self._set_params(AmplitudeA, mode, 'A', **kwargs)
-        self._setmaxorder(AmplitudeA, AmplitudeB)
-        if mode == ACMode.WHITENOISE:
-            self.Seed = kwargs.pop('Seed', datetime.now().timestamp())
-        self.PolynomA = numpy.zeros(self.MaxOrder+1)
-        self.PolynomB = numpy.zeros(self.MaxOrder+1)
-        ramps = kwargs.pop('Ramps', None)
-        if ramps is not None:
-            assert len(ramps)==4, \
-                'Ramps has to be a vector with 4 elements'
-            self.Ramps = ramps
-        super(VariableMultipole, self).__init__(family_name, **kwargs)
+        kwargs["Mode"] = kwargs.get("Mode", mode)
+        kwargs.setdefault("PassMethod", "VariableThinMPolePass")
+        if len(kwargs) > 2:
+            amp_aux = self._check_amplitudes(**kwargs)
+            for key, value in amp_aux.items():
+                if value is not None:
+                    kwargs["Amplitude" + key] = self._set_amplitude(value)
+                    kwargs = self._set_mode(mode, key, **kwargs)
+            maxorder = self._getmaxorder(amp_aux["A"], amp_aux["B"])
+            kwargs["MaxOrder"] = kwargs.get("MaxOrder", maxorder)
+            for key in amp_aux:
+                kwargs["Polynom" + key] = kwargs.get(
+                    "Polynom" + key, np.zeros(maxorder + 1)
+                )
+            ramps = self._check_ramp(**kwargs)
+            if ramps is not None:
+                kwargs["Ramps"] = ramps
+        super().__init__(family_name, **kwargs)
+
+    def _check_amplitudes(self, **kwargs: dict[str, any]) -> dict[str, any]:
+        amp_aux = {"A": None, "B": None}
+        all_amplitudes_are_none = True
+        for key in amp_aux:
+            if "Amplitude" + key in kwargs and "amplitude" + key in kwargs:
+                raise AtError("Duplicated amplitude " + key + "parameters.")
+            lower_case_kwargs = {k.lower(): v for k, v in kwargs.items()}
+            amp_aux[key] = lower_case_kwargs.pop("amplitude" + key.lower(), None)
+            if amp_aux[key] is not None:
+                all_amplitudes_are_none = False
+        if all_amplitudes_are_none:
+            raise AtError("Please provide at least one amplitude for A or B")
+        return amp_aux
+
+    def _set_amplitude(self, amplitude: float or _array or None) -> _array:
```

Comment:
_array is a function:
```python
def _array(value, shape=(-1,), dtype=numpy.float64): ...
```
You cannot use it as a type. Here, the type could be `float | Sequence[float]`